packages/tpetra/core/src/Tpetra_computeRowAndColumnOneNorms_def.hpp

```diff
@@ -187,13 +197,20 @@ computeLocalRowAndColumnOneNorms_RowMatrix (const Tpetra::RowMatrix<SC, LO, GO,
         if (! assumeSymmetric) {
           result_h.colNorms[lclCol] += matrixAbsVal;
         }
-      }
+      } // for each entry in row
 
+      // This is a local result.  If the matrix has an overlapping
+      // row Map, then the global result might differ.
```

Comment:
Would it make sense to the SIERRA use case to check for an overlapping row map before equilibration?

Reply:
@jhux2 Read the documentation :-) `Tpetra::computeRowAndColumnOneNorms` computes global results.  This comment appears in a function that computes a local result, in preparation for globalization.  See e.g., `globalizeRowOneNorms`.

Reply:
"Local" results do not account for overlap; "global" results do.  Sierra does not need to check for an overlapping row Map.  The `Tpetra::computeRowAndColumnOneNorms` function looks at the CrsMatrix's `Export` object; if it is trivial, there is no need to resolve overlap in row norms.

Reply:
The Sierra use case never exercises an overlapping row Map.  That would break MueLu and most Ifpack2 preconditioners.  I only implemented it because it's a straightforward generalization of column Map globalization.  The column Map is almost always overlapping, so it is almost always necessary to globalize the column Map.

"Globalize" means "resolve overlap."  That is, if Process 0 has an off-diagonal entry in global column C, and Process 1 also has an off-diagonal entry in global column C, the "local" routines won't detect both processes' contribution to the column norm for column C.  That's what `globalizeColumnOneNorms` does.

Reply:
@jhux2 To clarify: the answer to your question is "no."  `Tpetra::computeRowAndColumnOneNorms` checks whether row norms need globalization.  Sierra does not need to check.  Furthermore, the check is free; the function just gets the CrsMatrix's Export object pointer and checks whether the pointer is null.  If not null, then the CrsMatrix has a nontrivial Export object, so the row Map and range Map are not the same.